kanidmd/src/lib/core/https/mod.rs

```diff
@@ -94,6 +101,12 @@ impl RequestExtensions for tide::Request<AppState> {
             .map(|s| s.to_string())
             .map_err(|_| tide::Error::from_str(tide::StatusCode::ImATeapot, "teapot"))
     }
+
+    fn new_eventid(&self) -> (Uuid, String) {
```

Comment:
Because we use this as:

```
req.new_eventid()
```

To call something.function(), it always has to have &self as a parameter. 

Since this is a RequestExtension from the trait that is bound to `tide::Request<AppState>` (line 53 of https/mod.rs). We could omit (&self) but then we'd need to call it as `tide::Request<AppState>::new_eventid()` and I'm not even sure that would work.

The *intent* by making this a request extension and leaving self is later to allocate the auditscope inside the request, and then this will need self anyway.

But even if not, needing &self is the rule in rust if you want to call it against an instance, so it has to stay.

Reply:
> The _intent_ by making this a request extension and leaving self is later to allocate the auditscope inside the request, and then this will need self anyway.

Does this mean the `AuditScope` would be created in the `new_eventid` function? Or that it needs to know about the eventid somehow when it's created? The reason I asked was because it didn't make sense why you would need a `tide::Request::<AppState>` object to create something that has shares no information with it.

Reply:
It would be different. There would be a middleware on the request that creates the auditscope, and then new_eventid would query the request internals for "whats my current audit scope id" to return. But regardless, the self has to be there, because how else do you call the associated function?